Objects/Objects/Geometry/Line.cs

```diff
@@ -14,15 +14,16 @@ public class Line : Base, ICurve, IHasBoundingBox, ITransformable<Line>
 {
   public Line() { }
 
-  public Line(double x, double y, double z = 0, string units = Units.Meters, string applicationId = null)
+  [Obsolete("Line should not use a constructor that only sets the start point. Deprecated in 2.18.", true)]
+  public Line(double x, double y, double z = 0, string units = Units.Meters, string? applicationId = null)
   {
     start = new Point(x, y, z);
-    end = null;
+    end = null; // constructor is marked obsolete because we don't want to allow setting a null end point here.
```

Comment:
Is it worth us supressing the CA warning on this line? In core, we've supressed warnings in Obsolete functions (since they are legacy code, and can't/shan't be fixed)